### PR TITLE
Migrate user dashboard and Explore pages to SPA

### PR DESCRIPTION
**Pull request by anshg1214**
This PR migrates the User Dashboard and Explore pages to SPA

### CODE REVIEW

File: listenbrainz/webserver/views/explore.py
Comment (MonkeyDo) [inline listenbrainz/webserver/views/explore.py:26]: Don't know how relevant it is, but in some places (specifically API endpoints IIRC) we've had to add the `OPTIONS` method.
No idea if that would be needed here for some reason or not, just voicing it just in case.

**Review by MonkeyDo — COMMENTED** [review]
As you said, a big PR but I understand why it all needs to be done in one go…
Nice job it's working pretty smoothly!

I did notice an issue:
On the top-$entity component, switching entity type of time range result in replacing the whole content for some reason. I don't know why that i, but it's not idea.
In comparison, the "taste" page allows for seamless navigation between loved and hated tracks without re-rendering everything.


Everything else I tested works really well, great work ! :)

File: frontend/js/src/routes/index.tsx
Comment (anshg1214) [reply frontend/js/src/routes/index.tsx:47]: Yes, I need to rewrite UserFeedPage component, as it does not use props anywhere before removing this empty array.

File: frontend/js/src/user/stats/UserReports.tsx
Comment (anshg1214) [reply frontend/js/src/user/stats/UserReports.tsx:223]: I don't think we need it here. We're not passing current user as props as we're loading the global statistics first.

File: webpack.config.js
Comment (anshg1214) [reply webpack.config.js:110]: True, i've removed the file

File: listenbrainz/webserver/views/explore.py
Comment (anshg1214) [reply listenbrainz/webserver/views/explore.py]: Nope, fixed

File: frontend/js/src/user/year-in-music/2023/YearInMusic2023.tsx
Comment (anshg1214) [reply frontend/js/src/user/year-in-music/2023/YearInMusic2023.tsx:1518]: Sure! I'll do it in a followup PR

File: frontend/js/src/user/charts/UserEntityChart.tsx
Comment (anshg1214) [reply frontend/js/src/user/charts/UserEntityChart.tsx]: Fixed

File: frontend/js/src/user/charts/UserEntityChart.tsx
Comment (anshg1214) [reply frontend/js/src/user/charts/UserEntityChart.tsx]: Fixed

**Review by MonkeyDo — COMMENTED** [review]
Looking better and better! I think we're just about ready to merge after this last pass.

As we both discussed in private messages, I think after merging this we'll want to look at our bundling and code-splitting strategy because this index.js file is going to be pretty huge now :)

File: frontend/js/src/explore/cover-art-collage/SEO.tsx
Comment (anshg1214) [reply frontend/js/src/explore/cover-art-collage/SEO.tsx]: Fixed

File: frontend/js/src/explore/Explore.tsx
Comment (anshg1214) [reply frontend/js/src/explore/Explore.tsx]: This is good! I have added a default title and template.

**Review by MonkeyDo — COMMENTED** [review]
Took a while to test everything, but i only found a couple of issues, the rest is working great !

File: frontend/js/src/explore/routes/index.tsx
Comment (anshg1214) [reply frontend/js/src/explore/routes/index.tsx:30]: The `Outlet` component is used as a placeholder for child routes to render their elements into.

The children array of this route configuration contains three child routes. When one of these child routes is matched by the URL, the corresponding component will be rendered in place of the Outlet component.

File: frontend/js/src/explore/similar-users/SimilarUsers.tsx
Comment (anshg1214) [reply frontend/js/src/explore/similar-users/SimilarUsers.tsx:22]: Fixed.

Strange that `tbody` is not present in the current HTML file. https://github.com/metabrainz/listenbrainz-server/blob/22edab80736c4fc77737b0b5f9614ade75e8d89b/listenbrainz/webserver/templates/explore/similar-users.html#L23-L35

File: frontend/js/src/user/year-in-music/2021/YearInMusic2021.tsx
Comment (anshg1214) [reply frontend/js/src/user/year-in-music/2021/YearInMusic2021.tsx:988]: Fixed

File: frontend/js/src/error/ErrorBoundary.tsx
Comment (anshg1214) [reply frontend/js/src/error/ErrorBoundary.tsx]: Fixed

File: frontend/js/src/routes/routes.tsx
Comment (anshg1214) [reply frontend/js/src/routes/routes.tsx]: Fixed

File: frontend/js/src/error/ErrorBoundary.tsx
Comment (anshg1214) [reply frontend/js/src/error/ErrorBoundary.tsx:22]: I agree, Lets refactor this Error Boundary in the follow-up PR. We also have to rewrite the [this error boundary](https://github.com/metabrainz/listenbrainz-server/blob/22edab80736c4fc77737b0b5f9614ade75e8d89b/frontend/js/src/utils/ErrorBoundary.tsx).

File: frontend/js/src/explore/routes/index.tsx
Comment (MonkeyDo) [reply frontend/js/src/explore/routes/index.tsx:30]: Thanks !
Now I get it

**Review by MonkeyDo — APPROVED** [review]
Looking great @anshg1214 !

I deployed to tests again for testing, everything I tried is working fine.
Great job!

**Pull request merged**
